x-pack/plugins/enterprise_search/public/applications/analytics/components/add_analytics_collections/add_analytics_collection_logic.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { AnalyticsCollection } from '../../../../../common/types/analytics';
+import { HttpError, Status } from '../../../../../common/types/api';
+
+import { isAlphaNumericOrUnderscore } from '../../../../../common/utils/is_alphanumeric_underscore';
+import { generateEncodedPath } from '../../../shared/encode_path_params';
+import { flashAPIErrors, flashSuccessToast } from '../../../shared/flash_messages';
+import { KibanaLogic } from '../../../shared/kibana';
+import { AddAnalyticsCollectionsAPILogic } from '../../api/add_analytics_collection/add_analytics_collection_api_logic';
+import { COLLECTION_VIEW_PATH } from '../../routes';
+
+export interface AddAnalyticsCollectionsActions {
+  apiError(error: HttpError): HttpError;
+  apiSuccess(collection: AnalyticsCollection): AnalyticsCollection;
+  createAnalyticsCollection(): void;
+  makeRequest: typeof AddAnalyticsCollectionsAPILogic.actions.makeRequest;
```

Comment:
suggestion: This typing doesn't work, because the typeof makeRequest is an actionCreator rather than an action. If you check I believe you'll see that the makeRequest type is now `ActionsForPayloadBuilder<(args: unknown) => unknown>`, which isn't very helpful.

Reply:
Yeh! Thanks that solved that mystery. I should have checked out makeRequest myself.